playbooks/common/openshift-cluster/upgrades/v3_1_to_v3_2/pre.yml

```diff
@@ -110,8 +110,8 @@
   # are modified to use the correct image tag.  However, this can trigger a
   # docker restart if new configuration is laid down which would immediately
   # pull the latest image and defeat the purpose of these tasks.
-  - openshift_cli
-  pre_tasks:
+  - openshift_docker
```

Comment:
> Hmm, am I following you correctly:

> All of this would still take place in pre.yml. We already have a play for first master, all masters, all nodes, and all nodes + all masters.

Without breaking this into additional stages, yes (which I think is a really bad idea right now, but probably is a good idea for future upgrade work).

> I'm assuming this would be adding a second, new play for just the first master, it would immediately follow the existing one for first master, it would apply docker openshift role and query new versions.

It probably could be merged into the same play. The following oo_nodes_to_config play can be consolidated into the oo_masters_to_config:oo_nodes_to_config play, since we require masters be nodes ( to be extra safe, could conditionalize it on `when: 'node' in openshift`. 

> Then add a second new play for all hosts minus first master, which follows all existing verify steps, includes openshift_docker role, and then uses the version from the master.

If it's easier to just run it against all hosts, then that is fine as well, but it would save a small amount of time to exclude oo_first_master

> So we'd still be modifying docker related things in pre.yml, just later in the process, and we'd gain a little more confidence that everybody will be trying to use the same version?

Yeah, we would gain a canary for the docker changes on the master and we will avoid checking available versions on all hosts.

One thing that we might have to keep in mind is if a user is re-running a failed upgrade. I know @brenton and @sdodson have done some discussion here, but I'm not sure what the current state of handling for re-running upgrades is in place.